en/developer_guide/bbcode.rst

```diff
@@ -0,0 +1,172 @@
+            parameters:
+                use_option: true
+
+To declare that a tag uses option, you must add the key ``parameters.use_option`` and set it to true.
+Not that when declaring a tag with option, that option becomes mandatory. So if the option is not
+mandatory, you should declare two versions of the tag: one with option and one without. That way
+the parser will use one or the other according to the presence of an option. The ``html`` key uses
+both {param} and {option} string to indicate where to place those informations.
+The resulting BBCode will be ``[h={option}]{param}[/h]``
+
+Create a BBCode tag via a service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+BBCode definitions can also be declared by services. A compiler pass has been added to the
+BBCodeBundle to collect all services tagged with ``open_orchestra_bbcode.code_definitions``. Such
+a service includes a BBCode definition collection and serve it via the method ``getDefinitions``.
+So you are free to code your own service, but it must implement
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionCollectionInterface``.
+
+The definition collection must be composed of formatted elements implementing
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionInterface``. If your definition is a simple
+regex replacement, as the one seen in the previous section, you can use an Open Orchestra definition
+factory. But to create a BBCode with a more complex logic, you will have to create your own
+definition class. Both ways are explained in the next sections.
+
+Create simple definitions with the definition factory
+*****************************************************
+You can quickly generate simple definitions using the OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionFactory.
+This factory can be accessed via the service named ``open_orchestra_bbcode.definition.factory``
+
+To generate a new simple definition with this factory, call the method ``create`` with the following
+parameters:
+
+- ``$tag``: the BBCode tag, ie ``b`` for the tag [b]
+
+- ``$html``: the html representation of the tag, ie ``<strong>{param}</strong>`` for the [b]tag
+
+- ``$use_option``: whether the tag uses option or not
+
+So to create a bold tag definition object, you can write:
+
+.. code-block:: php
+
+    $definition = $container->get('open_orchestra_bbcode.definition.factory')
+        ->create('b', '<strong>{param}</strong>');
+
+And to create a header tag definition:
+
+.. code-block:: php
+
+    $definition = $container->get('open_orchestra_bbcode.definition.factory')
+        ->create('h', '<h{option}>{param}</h{option}>', true);
+
+As mentionned before, to let Open Orchestra know your new definition, you have to add it to a
+definition collection class implementing
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionCollectionInterface``. Then make that
+collection a Symfony service tagged with ``open_orchestra_bbcode.code_definitions``.
```

Comment:
I don't know what to show: it's straight forward as the interface only have one explicit method: getDefinitions. I think it may unnecessarily burden the explanation :s